pandas/tests/indexes/test_indexing.py

```diff
@@ -248,6 +249,20 @@ def test_putmask_with_wrong_mask(self, index):
             index.putmask("foo", fill)
 
 
+class TestCasting:
+    def test_maybe_cast_with_dtype(self):
+        # https://github.com/pandas-dev/pandas/issues/32413
+        frame = DataFrame(index=Index([1, np.nan]))
+
+        cond = frame.index.notna()
+        other = "a" + frame.reset_index().index.astype(str)
+
+        fixed_index = frame.index.where(cond, other)
+
+        assert (other == Index(["a0", "a1"])).all()
```

Comment:
Generally should be asserted using `tm.assert_index_equal(result, expected)`

Reply:
Thanks! I've changed the test to use this.